### DISCUSSION TITLE
multiple hardware event support

### DISCUSSION THREAD

**Discussion by hilldani** [async-profiler/async-profiler#824] [Q&A] [answered]
I know async profiler can generate stacks from a single hardware event but sometimes you need multiple hardware events to generate a valuable flamegraph. For example in perf we can do `perf record -a -g -e cycles,instructions sleep 2` and then generate a [CPI flamegraph](https://www.brendangregg.com/blog/2014-10-31/cpi-flame-graphs.html) (cycles per instruction, essentially how efficiently instructions are flowing through the cpu). However I cannot figure out how to do this with async-profiler. Is my only option to use perf-map-agent?

**Comment by apangin** [comment, marked as answer]
Async-profiler can currently collect only one perf event at a time.
Feel free to open a feature request. This won't be a high priority feature though.
